pkg/minikube/assets/addons.go

```diff
@@ -770,6 +771,22 @@ func SelectAndPersistImages(addon *Addon, cc *config.ClusterConfig) (images, cus
 	return images, customRegistries, err
 }
 
+// fixes addon image names according to image repository used
+func fixAddonImages(repo string, images map[string]string) map[string]string {
+	if repo == "registry.cn-hangzhou.aliyuncs.com/google_containers" {
+		// for aliyun registry must strip namespace from image name, e.g.
+		//   registry.cn-hangzhou.aliyuncs.com/google_containers/k8s-minikube/storage-provisioner:v5 will not work
+		//   registry.cn-hangzhou.aliyuncs.com/google_containers/storage-provisioner:v5 does work
+		newImages := make(map[string]string)
+		for name, image := range images {
+			tmp := strings.Split(image, "/")
```

Comment:
instead of splitting the image here, could you just use `strings.TrimPrefix(image, "k8s-minkube/")`? this will prevent accidentally trimming something we don't want to.